detectors/gcp/detector.go

```diff
@@ -0,0 +1,135 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp // import "go.opentelemetry.io/contrib/detectors/gcp"
+
+import (
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/compute/metadata"
+	"github.com/GoogleCloudPlatform/opentelemetry-operations-go/detectors/gcp"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.10.0"
+)
+
+// NewDetector returns a resource detector which detects resource attributes on:
+// * Google Compute Engine (GCE)
+// * Google Kubernetes Engine (GKE)
+// * Google App Engine (GAE)
+// * Cloud Run
+// * Cloud Functions
+func NewDetector() resource.Detector {
+	return &detector{detector: gcp.NewDetector()}
+}
+
+type detector struct {
+	detector gcpDetector
+}
+
+// Detect detects associated resources when running on GCE, GKE, GAE,
+// Cloud Run, and Cloud functions.
+func (d *detector) Detect(ctx context.Context) (*resource.Resource, error) {
+	if !metadata.OnGCE() {
+		return nil, nil
+	}
+	projectID, err := d.detector.ProjectID()
+	if err != nil {
+		return nil, err
+	}
+	attributes := []attribute.KeyValue{semconv.CloudProviderGCP, semconv.CloudAccountIDKey.String(projectID)}
+
+	switch d.detector.CloudPlatform() {
+	case gcp.GKE:
+		attributes = append(attributes, semconv.CloudPlatformGCPKubernetesEngine)
+		v, locType, err := d.detector.GKEAvailabilityZoneOrRegion()
+		if err != nil {
+			return nil, err
```

Comment:
Based on [this](https://github.com/open-telemetry/opentelemetry-go/blob/a50cf6aadd582f9760c578e2c4b5230b6c30913d/sdk/resource/auto.go#L35-L40):

> 	// Detect returns an initialized Resource based on gathered information.
>	// If the source information to construct a Resource contains invalid
>	// values, a Resource is returned with the valid parts of the source
>	// information used for initialization along with an appropriately
>	// wrapped ErrPartialResource error.

Shouldn't this return a partial resource and a wrapped `ErrPartialResource` in these error cases?